### PR TITLE
Added copy, delete, rename, move object browser context menu options

### PR DESCRIPTION
**Pull request by Wright4i**
### Changes

Added Copy, Delete, Rename, and Move to Object Browser object context options - Issue #826 
image

#### Copy
image
image

#### Delete
image

#### Rename
image
image

#### Move
image
image
image

### Checklist

* [x] have tested my change
* [ ] updated relevant documentation
* [ ] Remove any/all `console.log`s I added
* [x] eslint is not complaining
* [ ] have added myself to the contributors' list in [CONTRIBUTING.md](https://github.com/halcyon-tech/vscode-ibmi/blob/master/CONTRIBUTING.md)
* [x] **for feature PRs**: PR only includes one feature enhancement.

### CODE REVIEW

File: src/views/objectBrowser.js [hunk 1]
```diff
@@ -681,8 +681,122 @@ module.exports = class objectBrowserTwoProvider {
...
+          let newPath = node.path;
+          let newPathOK;
+          do {
+            newPath = await vscode.window.showInputBox({
+              prompt: `Create duplicate object to new library/object`,
+              value: newPath,
+              validateInput: newPath => {
+                let splitPath = newPath.split(`/`);
+                if (splitPath.length != 2) return `Invalid path: ${newPath}. Use format LIB/OBJ`;
+                return newPath.length <= 21 ? null : `Object path must be 21 chars or less.`;
```
Comment (chrjorgensen) [inline src/views/objectBrowser.js:697]: One check is missing: Both parts must be 10 char or less...

**Comment by chrjorgensen** [comment]
Option to move object to another library would also be nice... 😉

File: src/views/objectBrowser.js (hunk 1 above)
Comment (Wright4i) [reply src/views/objectBrowser.js]: Fixed in [e09c220](https://github.com/halcyon-tech/vscode-ibmi/pull/882/commits/e09c220a90b693b5644724f7041983c76cd6570f)

image
image

**Comment by Wright4i** [comment]
> Option to move object to another library would also be nice... 😉

Good idea! Added in [e628d73](https://github.com/halcyon-tech/vscode-ibmi/pull/882/commits/e628d73317eff051ba7c1d89656451301e1f851a)

**Comment by chrjorgensen** [comment]
@Wright4i One final comment before I go to bed: Could you write the object name and type instead of just `object` in the information messages? The object may have been dropped from the list (if autorefresh) and there will be no sign of what object was being handled.

I will test this change tomorrow... looking good to me so far! 😃

**Comment by Wright4i** [comment]
@chrjorgensen object type now in the auto refresh info messages [4c8b566](https://github.com/halcyon-tech/vscode-ibmi/pull/882/commits/4c8b566b4cf988cd40f1e264c3ea2c9832536057). Thanks for taking the time! :)

**Comment by chrjorgensen** [comment]
@Wright4i Thanks for making this PR!

Could you write the object name as well? Like you do in the delete object function. Then the user can see the old name - and maybe regret... 😄

**Comment by Wright4i** [comment]
@chrjorgensen I think I caught that just before you commented it [a8048f9](https://github.com/halcyon-tech/vscode-ibmi/pull/882/commits/a8048f9c35159de65f906b00dccb3e348225592f)

**Comment by chrjorgensen** [comment]
@Wright4i I will test this tomorrow. Thanks!

File: src/views/objectBrowser.js [hunk 2]
```diff
@@ -681,8 +681,164 @@ module.exports = class objectBrowserTwoProvider {
...
+          //Running from command
+          console.log(this);
+        }
+      }),
+
+      vscode.commands.registerCommand(`code-for-ibmi.deleteObject`, async (node) => {
+
+        if (node) {
+          //Running from right click
+          let result = await vscode.window.showWarningMessage(`Are you sure you want to delete ${node.path}?`, `Yes`, `Cancel`);
```
Comment (chrjorgensen) [inline src/views/objectBrowser.js:735]: Please specify the object type in the warning message - more than one object in the library could exist with the name...

**Comment by chrjorgensen** [comment]
@Wright4i Looking good! Only one thing - see comment above.

File: src/views/objectBrowser.js (hunk 2 above)
Comment (Wright4i) [reply src/views/objectBrowser.js]: Done! [99a5136](https://github.com/halcyon-tech/vscode-ibmi/pull/882/commits/99a5136e47ed86e5ff17b58ee1be26212829ed25)

**Comment by chrjorgensen** [comment]
@Wright4i Great job! Let's have @worksofliam check, test and merge...

**Comment by worksofliam** [comment]
Tested all new commands and working great!

Thank you @Wright4i for the PR. Thank you @chrjorgensen for the additional testing and feedback :)

**Pull request merged**